snuba/replacers/errors_replacer.py

```diff
@@ -478,50 +475,106 @@ def get_timestamp_condition(msg_field: str, operator: str) -> str:
     manually excluded from search queries.
 
-    See docstring of process_exclude_groups for an explanation of how this is used.
+    See docstring of ExcludeGroupsReplacement for an explanation of how this is
+    used.
 
     Note that events merged this way cannot be cleanly unmerged by
     process_unmerge, as their group hashes possibly stand in no correlation to
     how the merging was done.
     """
 
-    event_ids = message.data["event_ids"]
-    if not event_ids:
-        return None
+    event_ids: Sequence[str]
+    project_id: int
+    from_timestamp: Optional[str]
+    to_timestamp: Optional[str]
+    new_group_id: int
+    replacement_type: ReplacementType
+    replacement_message_metadata: ReplacementMessageMetadata
+    all_columns: Sequence[FlattenedColumn]
+
+    @classmethod
+    def parse_message(
+        cls,
+        message: ReplacementMessage[ReplaceGroupMessageBody],
+        context: ReplacementContext,
+    ) -> Optional[ReplaceGroupReplacement]:
+        event_ids = message.data["event_ids"]
+        if not event_ids:
+            return None
 
-    event_set_filter = _build_event_set_filter(message.data, state_name)
-    if event_set_filter is None:
+        return cls(
+            event_ids=event_ids,
+            project_id=message.data["project_id"],
+            from_timestamp=message.data.get("from_timestamp"),
+            to_timestamp=message.data.get("to_timestamp"),
+            new_group_id=message.data["new_group_id"],
+            replacement_type=message.action_type,
+            replacement_message_metadata=message.metadata,
+            all_columns=context.all_columns,
+        )
+
+    def get_project_id(self) -> int:
+        return self.project_id
+
+    def get_query_time_flags(self) -> Optional[QueryTimeFlags]:
         return None
 
-    prewhere, where, query_args = event_set_filter
+    def get_replacement_type(self) -> ReplacementType:
+        return self.replacement_type
 
-    full_where = f"PREWHERE {' AND '.join(prewhere)} WHERE {' AND '.join(where)}"
-    project_id: int = message.data["project_id"]
-    query_time_flags = (None, project_id)
+    def get_message_metadata(self) -> ReplacementMessageMetadata:
+        return self.replacement_message_metadata
 
-    return _build_group_replacement(
-        message,
-        project_id,
-        full_where,
-        query_args,
-        query_time_flags,
-        all_columns,
-    )
+    @cached_property
+    def _where_clause(self) -> str:
+        prewhere, where, query_args = _build_event_set_filter(
+            project_id=self.project_id,
+            event_ids=self.event_ids,
+            from_timestamp=self.from_timestamp,
+            to_timestamp=self.to_timestamp,
+        )
+
+        return (
+            f"PREWHERE {' AND '.join(prewhere)} WHERE {' AND '.join(where)}"
+            % query_args
+        )
+
+    def get_count_query(self, table_name: str) -> Optional[str]:
```

Comment:
Curious about this interface - how come `table_name` isn't just a property on the replacement

Reply:
I did that just to satisfy typechecking, and I don't think it matters for runtime. those parameters are string-interpolated